cli/module_registry.go

```diff
@@ -0,0 +1,131 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+)
+
+// These types mirror app's internal representations of modules
+
+// ModuleVisibility determines whether modules are public or private.
+type ModuleVisibility string
+
+// Permissions enumeration.
+const (
+	ModuleVisibilityPrivate ModuleVisibility = "private"
+	ModuleVisibilityPublic  ModuleVisibility = "public"
+)
+
+// ModuleComponent represents an api - model pair.
+type ModuleComponent struct {
+	API   string `json:"api"`
+	Model string `json:"model"`
+}
+
+// ModuleManifest is used to create & parse manifest.json.
+type ModuleManifest struct {
+	Name        string            `json:"name"`
+	Visibility  ModuleVisibility  `json:"visibility"`
+	URL         string            `json:"url"`
+	Description string            `json:"description"`
+	Models      []ModuleComponent `json:"models"`
+	Entrypoint  string            `json:"entrypoint"`
+}
+
+const (
+	defaultManifestFilename = "meta.json"
+)
+
+// CreateModuleCommand runs the command to create a module
+// This includes both a gRPC call to register the module on app.viam.com and creating the manifest file.
+func CreateModuleCommand(c *cli.Context) error {
+	moduleNameArg := c.String("name")
+	orgIDArg := c.String("org_id")
+	publicNamespaceArg := c.String("public_namespace")
```

Comment:
I think the location of this function is the most controversial part of this PR.

I had originally placed it in `viam/main.go` however the size of that file made me hesitant to continue adding to it (especially because there will be more `module` sub commands). 

All of the other functions in `viam/main.go` are short and sweet wrappers of the APIs, however some of the module subcommands will include more logic on the cli side so it made sense for me to put them in here.

Happy to be wrong about this though.

Reply:
I like the code here honestly, because as you said the code is larger. Normally I don't like breaking convention in something and not following through with the rest, leaving 2 ways code is organized.

But in this case its so minor and i think having the logic be encasulated in separate files and main.go be just the command configuration, makes the most sense to me. This can be refactored later if we end up liking/disliking it.

Reply:
I like it

Reply:
Also this CLI needs a lot of work so I think it's okay to experiment with new ideas until we do a rewrite